File: javascript-solutions/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numMap = new Map();
    for (let i = 0; i < nums.length; i++) {
        const currentNum = nums[i];
        const missing = numMap.get(target - currentNum)
        if (missing !== undefined) {
            return [missing, i];
        }
        
        numMap.set(currentNum, i);
    }
}

File: go-solutions/twoSum.go
func twoSum(nums []int, target int) []int {
    for i:=0; i < len(nums); i++ {
        for j:=i+1; j < len(nums); j++ {
            //fmt.Println("nums[i]: " + strconv.Itoa(nums[i]) + " nums[j]: " + strconv.Itoa(nums[j]))
            if (nums[i] + nums[j] == target) {
                return []int{i,j}
            }
        }
    }
    return []int{0,0}
}
